src/pip/_internal/operations/install/wheel.py

```diff
@@ -563,28 +564,23 @@ def is_entrypoint_wrapper(name):
             logger.warning(msg)
 
     # Record pip as the installer
-    installer = os.path.join(dest_info_dir, 'INSTALLER')
-    temp_installer = os.path.join(dest_info_dir, 'INSTALLER.pip')
-    with open(temp_installer, 'wb') as installer_file:
+    installer_path = os.path.join(dest_info_dir, 'INSTALLER')
+    with adjacent_tmp_file(installer_path) as installer_file:
         installer_file.write(b'pip\n')
-    shutil.move(temp_installer, installer)
-    generated.append(installer)
+    replace(installer_file.name, installer_path)
+    generated.append(installer_path)
 
     # Record details of all files installed
-    record = os.path.join(dest_info_dir, 'RECORD')
-    temp_record = os.path.join(dest_info_dir, 'RECORD.pip')
-    with open_for_csv(record, 'r') as record_in:
-        with open_for_csv(temp_record, 'w+') as record_out:
-            reader = csv.reader(record_in)
-            outrows = get_csv_rows_for_installed(
-                reader, installed=installed, changed=changed,
-                generated=generated, lib_dir=lib_dir,
-            )
-            writer = csv.writer(record_out)
-            # Sort to simplify testing.
-            for row in sorted_outrows(outrows):
-                writer.writerow(row)
-    shutil.move(temp_record, record)
+    record_path = os.path.join(dest_info_dir, 'RECORD')
+    with open(record_path, **csv_io_kwargs('r')) as record_file:
+        # Sort to simplify testing.
+        rows = sorted_outrows(get_csv_rows_for_installed(
+            csv.reader(record_file), installed=installed, changed=changed,
+            generated=generated, lib_dir=lib_dir))
```

Comment:
This isn't super readable IMO. I suggest breaking this into two statements, and changing to a different style for arguments (one per line).

```suggestion
        csv_rows = get_csv_rows_for_installed(
            csv.reader(record_file),
            installed=installed,
            changed=changed,
            generated=generated,
            lib_dir=lib_dir,
        )
        rows = sorted_outrows(csv_rows)
```

Reply:
This has been addressed in f164531 with hanging indentation (I've seen this style used in `pip`, so hopefully it's fine).